Tests/SwiftStdlibTests/ArrayExtensionsTests.swift

```diff
@@ -36,6 +36,21 @@ final class ArrayExtensionsTests: XCTestCase {
         XCTAssertEqual(swappedEmptyArray, emptyArray)
     }
 
+    func testSortedLike() {
+        let expected1 = [1, 2, 3, 4, 5]
+        let candidate1 = [2, 5, 3, 1, 4]
+        XCTAssertEqual(candidate1.sorted(like: expected1, keyPath: \.self), expected1)
+
+        let candidate2 = [2, 5, 3, 6, 1, 4]
+        XCTAssertEqual(candidate2.sorted(like: expected1, keyPath: \.self), [1, 2, 3, 4, 5, 6])
```

Comment:
If we remove the `5` value from the expected(btw seems not the correct name), can we guarantee that the order is stable? I mean the result is always  `[1, 2, 3, 4, 5, 6]` or it could be `[1, 2, 3, 4, 6, 5])` ?

Reply:
You are correct, values not present in the ordered array are sorted last and are not stable.

Reply:
It's not completely unstable, it just orders them as they are after the sorted elements, no?

Reply:
I think it depends on the implementation of the sorting algorithm, `sorted(by:)` in this case.